src/core/core.cpp

```diff
@@ -792,10 +793,19 @@ void Core::setUsername(const QString& username)
 
 void Core::setAvatar(const QByteArray& data)
 {
-    QPixmap pic;
-    pic.loadFromData(data);
-    Settings::getInstance().saveAvatar(pic, getSelfId().toString());
-    emit selfAvatarChanged(pic);
+    if (data != nullptr)
+    {
+        QPixmap pic;
+        pic.loadFromData(data);
+        Settings::getInstance().saveAvatar(pic, getSelfId().toString());
+        emit selfAvatarChanged(pic);
+    }
+    else
+    {
+        QString selfPubKey = getSelfId().publicKey;
+        QFile::remove(Settings::getInstance().getSettingsDirPath()+"avatars/"+selfPubKey.left(64)+".png");
```

Comment:
You probably should check whether `QFile::remove` returns `true`, and if it isn't, warn user that it was not possible to remove avatar.

Also, avatar doesn't change if it wasn't possible to remove it ↓